### DISCUSSION TITLE
Adding a second physics module does not work

### DISCUSSION THREAD

**Discussion by tctheodosiou** [idaholab/moose#25184] [Q&A Getting Started] [answered]
Hello everyone,
I am trying to find my way in MOOSE following the Tutorials. I completed the TensorMechanics Module, and proceeded to Contact. My problem is that the app is not running; I get the following error:
_... section '[Contact]' does not have an associated "Action"._

Steps sor far:
1. I created an app using stork
2. Enabled the TensorMechanics module and performed "make".
3. I followed all steps of the TensorMechanics Tutorial - so far so good.
4. I enabled the Contact module in the Makefile and performed "make" again. It compile without problems. (If I run make again, it makes no new compilations, so I guess that the compilation has been successful).
5. I follow the Contact tutorial (https://mooseframework.inl.gov/modules/contact/tutorials/introduction/step01.html). I even copy/pasted the code to make sure that there is no typo in my code.
6. I get the error mentioned above (_... section '[Contact]' does not have an associated "Action"._)

Furthermore, VS Code does not recognise Contact for AutoCompletion; not even for the "[Contact]" (Autocomplete works fine for TensorMechanics).

What am I doing wrong? How shoule I "inform" the app that I need a new module in my app?
Any advice is most welcome!!

Thank you

Update: I tries to "process" the app using peacock. I get the following warning in the console: 
_WARNING
 Could not create /Contact_

**Comment by GiudGiud** [comment, marked as answer]
Hello

TensorMechanics does not depend on Contact, Contact depends on TM.
So you can run TM simulations with the Contact module executable, but no Contact simulation with the TM executable

So you need to enable both, like you did in 2 then in 4, unfortunately you built in 2, and it will not pick it up until you clean and rebuild.

so 
```
make clobberall
make
```
in your storked application

Guillaume

**Reply by tctheodosiou** [reply]
Hi, Thank you for your prompt response!

When I rebuilt (Step 4), the executable had a recent creation date (few seconds ago), so I assumed that a correct re-compilation took place.

I did an update of moose (git fetch origin, rebase origin/master), and then make. It recompiled and worked without out of the box.
I guess that when I updated the code, it was forced to recompile everything, as you suggested.

Now everything works as expected.
